modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,288 @@
+
+[NOTE]
+====
+The clean up process relies on a Python virtual environment to ensure
+that the python packages version that you use supports tags for Octavia
+objects. If you are certain that your environment uses a minium `openstacksdk`
+version 0.54.0, `python-openstackclient` version 5.5.0 and
+`python-octaviaclient` version 2.3.0, you can forego the virtual environment.
+====
+
+.Prerequisites
+
+* You installed the {product-title} CLI, `oc`.
+* You installed a Python interpreter.
+* You installed the `openstacksdk` Python package.
+* You installed the `openstack` CLI.
+* You have access to the underlying {rh-openstack} cloud.
+* You can access the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+. Create a clean-up Python virtual environment:
+.. Create a temporary directory for your environment. For example, with `venv`:
++
+[source,terminal]
+----
+$ python3 -m venv /tmp/venv
+----
++
+`venv` is used in all clean up examples.
+.. Enter the virtual environment. For example:
++
+[source,terminal]
+----
+$ source /tmp/venv/bin/activate
+----
+.. Upgrade `pip` in the virtual environment by running the following command:
++
+[source,terminal]
+----
+(venv) $ pip install pip --upgrade
+----
+.. Install the required Python packages by running the following command:
++
+[source,terminal]
+----
+(venv) $ pip install openstacksdk==0.54.0 python-openstackclient==5.5.0 python-octaviaclient==2.3.0
+----
+
+. On a command line, enter the following commands to set variables to cluster and Kuryr identifiers:
+
+.. Set the cluster ID:
++
+[source,terminal]
+----
+(venv) $ CLUSTERID=$(oc get infrastructure.config.openshift.io cluster -o=jsonpath='{.status.infrastructureName}')
+----
+
+.. Set the cluster tag:
++
+[source,terminal]
+----
+(venv) $ CLUSTERTAG="openshiftClusterID=${CLUSTERID}"
+----
+.. Set the router ID:
++
+[source,terminal]
+----
+(venv) $ ROUTERID=$(oc get kuryrnetwork -A --no-headers -o custom-columns=":status.routerId"|head -n 1)
+----
+
+. To create a Bash function that removes finalizers from specified resources, enter the following command:
++
+[source,terminal]
+----
+(venv) $ function REMFIN {
+    local resource=$1
+    local finalizer=$2
+    for res in $(oc get $resource -A --template='{{range $i,$p := .items}}{{ $p.metadata.name }}|{{ $p.metadata.namespace }}{{"\n"}}{{end}}'); do
+        name=${res%%|*}
+        ns=${res##*|}
+        yaml=$(oc get -n $ns $resource $name -o yaml)
+        if echo "${yaml}" | grep -q "${finalizer}"; then
+            echo "${yaml}" | grep -v  "${finalizer}" | oc replace -n $ns $resource $name -f -
+        fi
+    done
+}
+----
++
+The function takes two parameters: the first parameter is name of the resource, and the second parameter is the
+finalizer to remove. The named resource is removed from the cluster and its definition is replaced with copied data excluding the specified finalizer.
+
+. To remove Kuryr finalizers from services, enter the following command:
```

Comment:
It would be long nevertheless, even if I place all of the cleaning commands under one "and now let's clean those up!" section ;). All the steps are necessary, unfortunately.

Reply:
Yeah, I was just trying to think of a possible way to clean up the clean up section 😄. But you're right, its a very long procedure anyway